mongodb-atlas/src/main/java/io/vantiq/atlasConnector/AtlasStorageMgr.java

```diff
@@ -98,8 +98,13 @@ public Completable initialize(Vertx vertx) {
             return Flowable.fromPublisher(adminDb.runCommand(new Document("ping", 1)))
                     .doOnSubscribe(s -> start.set(System.currentTimeMillis()))
                     .map(doc -> {
-                if (doc.get("ok", 0) != 1) {
-                    throw new Exception("Failed to ping MongoDB Atlas");
+                // depending on the server version: the result can be an Integer 1, and it can be a Double 1.0
+                if (doc.containsKey("ok")) {
+                    if (doc.get("ok") instanceof Integer && (Integer)doc.get("ok") != 1) {
+                        throw new Exception("Failed to ping MongoDB Atlas");
+                    } else if (doc.get("ok") instanceof Double && (Double)doc.get("ok") != 1.0) {
+                        throw new Exception("Failed to ping MongoDB Atlas");
+                    }
```

Comment:
Is it possible for the doc to not contain "ok" and is it possible for the "ok" value to be anything else then an Integer or Double? Wondering if the logic is missing a possible failure scenario.

Reply:
maybe we can just run with the fact that the command returned w/o error, and not inspect the result document? The documentation says nothing about what the result is supposed to contain.